.github/workflows/release.yaml

```diff
@@ -0,0 +1,112 @@
+# Copyright (c) 2020-2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+name: "NeMo Aligner release"
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch: 
+        description: Branch to release
+        required: true
+        type: string
+jobs: 
+  main:
+    if: contains(fromJSON('["terryk", "gshennvm", "ko3n1g"]'), github.actor)
+    runs-on: ubuntu-latest
+    environment: 
+      name: main
+    steps:  
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          path: ${{ github.run_id }}
+          ref: ${{ inputs.branch }}
+
+      - name: Create release
+        id: version-number
+        run: |
+          cd ${{ github.run_id }}
+          VERSION=$(python -c "import nemo_aligner; print(nemo_aligner.__version__)")
+          
+          NAME="NVIDIA NeMo-Aligner ${VERSION}"
+          CHANGELOG=$(awk '/^## '"$NAME"'/{flag=1; next} /^## /{flag=0} flag' CHANGELOG.md)
+          CHANGELOG=$(echo "$CHANGELOG" | sed '/./,$!d' | sed ':a;N;$!ba;s/\n$//')
+
+          PAYLOAD=$(jq \
+                      -n \
+                      -c \
+                      --arg CI_COMMIT_BRANCH "${{ inputs.branch }}" \
+                      --arg NAME "$NAME" \
+                      --arg BODY "$CHANGELOG" \
+                      '{
+                        "tag_name": $CI_COMMIT_BRANCH,
```

Comment:
Does this need to be a real tag? I tried creating in the UI and it looks like it didn't accept a branch for the tag field

Reply:
It creates a tag based on the current sha of the specified branch:)

Reply:
oh nice!